### DISCUSSION TITLE
pnpm - js package manager

### DISCUSSION THREAD

**Discussion by alix1383** [NativePHP/laravel#296] [Ideas]
I think it is a good idea to install js deps with pnpm.

**Comment by simonhamp** [comment]
You can already use the `--installer` flag on the `native:install` and `native:serve` commands so you can use `yarn` or `pnpm`:

```shell
php artisan native:install --installer=pnpm
```

Maybe this needs to be extended to the `native:build` and `native:publish` commands too? Could be a nice PR for a first issue.

**Reply by simonhamp** [reply]
Also, I'm not sure if anyone is using `bun` but it might be cool to add that too?

**Reply by alix1383** [reply]
I sure many people using bun. 
Also it's better than others js runtimes

**Comment by alix1383** [comment]
Sorry for such a question, I have never created a Composer package before.

I want to make some changes.

Can you show me how I can use a fork of the Electron repository in my composer.json file? 

I tried something, but it didn't work.

**Reply by simonhamp** [reply]
Here's a really [concise answer on StackOverflow](https://stackoverflow.com/a/38698049/123696).

Once you've updated your `composer.json`, you need to run `composer update`.

**Comment by binaryfire** [comment]
Having it as an option is cool, but please don’t make pnpm the default. There are so many annoying edge cases with pnpm, eg. when working with Docker containers on Macs. Their GitHub issues are full of examples.

NPM “just works” and is much faster that it used to be so IMHO that should always be the default.

the `—installer` flag is cool though

**Reply by simonhamp** [reply]
Don't worry, it definitely won't be the default. `npm` is by far the most widely used so that will remain the default.

What might be good is that you could set the default in your `.env` so you don't have to keep on passing the `--installer` flag.
